platform-api/src/main/java/io/datacater/core/deployment/DeploymentEndpoint.java

```diff
@@ -341,16 +347,17 @@ private List<String> getDeploymentLogsAsList(UUID deploymentId) {
     return Arrays.asList(k8Deployment.getLogs(deploymentId).split("\n"));
   }
 
-  private HttpRequest buildDeploymentServiceRequest(UUID deploymentId, String path) {
+  private HttpRequest buildDeploymentServiceRequest(UUID deploymentId, String path, int replica) {
     K8Deployment k8Deployment = new K8Deployment(client);
-    String clusterIp = k8Deployment.getClusterIp(deploymentId);
+    String ip = k8Deployment.getDeploymentReplicaIp(deploymentId, replica).replace(".", "-");
+    String service = StaticConfig.SERVICE_NAME_PREFIX + deploymentId;
+    String namespace = StaticConfig.EnvironmentVariables.NAMESPACE;
+    int port = StaticConfig.EnvironmentVariables.DEPLOYMENT_CONTAINER_PORT;
+    String protocol = StaticConfig.EnvironmentVariables.DEPLOYMENT_CONTAINER_PROTOCOL;
+
     String uriReady =
         String.format(
-            "%s://%s:%d%s",
-            StaticConfig.EnvironmentVariables.DEPLOYMENT_CONTAINER_PROTOCOL,
-            clusterIp,
-            StaticConfig.EnvironmentVariables.DEPLOYMENT_CONTAINER_PORT,
-            path);
+            "%s://%s.%s.%s.svc.cluster.local:%d%s", protocol, ip, service, namespace, port, path);
```

Comment:
I feel that we could get rid of the service here. From my understanding, we could [directly access the replica via its pod IP](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pods).

Reply:
I think that we no longer need it given that we are directly working with pods. What do you think?